### ISSUE TITLE
required attribute on md-select not working anymore

### ISSUE THREAD

**Issue by wembernard** [angular/material#7067]
I guess it's related to #6393 but @DevVersion didn't seem to fix md-select (only md-autocomplete), so I'm opening a new issue. 

I've been trying 
```
      <md-input-container class="md-block">
        <label>Favorite Number</label>
        <md-select name="myModel" ng-model="myModel" required="true">
          <md-option value="1">One</md-option>
          <md-option value="2">Two</md-option>
        </md-select>
      </md-input-container>
```
with angular-material#1.0.3 and angular-material#1.0.5 but didn't make it work. 

Some [Demo](http://plnkr.co/edit/38W2pKpD2Tgw4KteGdLS?p=preview)

Any idea/news on this problem?

**Comment by DevVersion** [comment]
@wembernard Thanks for filing a new issue. I will implement that too :+1:

**Comment by DevVersion** [comment]
@wembernard 
Waiting for a response [here](https://github.com/angular/material/pull/6803#issuecomment-181365826)

> At the moment in this PR, in most cases we support required="false" required="true", so the required attribute is accepting a boolean as value. But this is not a default behavior in HTML from W3C.

> The users should better use ng-disabled, ng-readonly and ng-required.
So IMO we should not parse the attribute values from these type of attributes.

> - disabled
- readonly
- required

So I'm not sure if I really should implement the new parses for these type of attributes.

**Comment by wembernard** [comment]
@DevVersion ng-required didn't work either.

Here is a [Demo](http://plnkr.co/edit/38W2pKpD2Tgw4KteGdLS?p=preview)

**Comment by DevVersion** [comment]
@wembernard Thanks for the demo. I will try to take a look at the code.

**Comment by DevVersion** [comment]
@wembernard In general `ng-required` works. As you can see here in the demo (http://codepen.io/DevVersion/pen/OMaBBY?editors=1010) `ng-required` is accepting boolean scope variables. But the thing is, the form will be submitted (in case that the form is invalid). Is that your actual problem?

**Comment by DevVersion** [comment]
@wembernard As native components (like `input` etc.) will prevent form from being submitted, you should add to your form: `name="myForm" ng-submit="myForm.$valid && ctrl.submit()"`.

**Comment by wembernard** [comment]
Thanks @DevVersion for looking into this.

The form from being submitted is indeed one of my problems. I guess whether it's a simple input or a complex md-select, submitting forms should behave the same way: either block the submission or send it. I'm ok using `myForm.$valid` but it should be needed for `<input>` too in this case. 

The second one is the display: it should be red to inform user one of the validation on this field did not pass. Just like a simple <input>, right?

**Comment by DevVersion** [comment]
@wembernard Yes definitely, it should be red. Extra for this case was `ngMessages` built. As you can see in following example, if you open the panel (which sets the select to touched ) and then close it without selecting an item, you will the ngMessage will appear. Another example is, select a value and then set the select model to undefined (using the clear button)

http://codepen.io/DevVersion/pen/OMaBBY?editors=1010

**Comment by wembernard** [comment]
@DevVersion I added the simple input to note behaviour differences: http://codepen.io/anon/pen/qbLJvV?editors=1010

**Comment by DevVersion** [comment]
@wembernard Yes, I know, the native input with `required` looks much more user-friendly.

**Comment by wembernard** [comment]
Actually, md-select used to work the same:

<img width="867" alt="screen shot 2016-02-15 at 09 39 45" src="https://cloud.githubusercontent.com/assets/11645044/13043389/370e7c74-d3c8-11e5-9dc5-a51d874d8f99.png">
(from https://material.angularjs.org/latest/demo/select)

**Comment by DevVersion** [comment]
@wembernard I'll close that issue because the attribute in general works. (required` and `ng-required`)
- See select's attribute validation - http://codepen.io/DevVersion/pen/OMaBBY?editors=1010
- See **required** select in form - http://codepen.io/DevVersion/pen/eZYRQL

About the form validation (errors when submitting form) there was already an issue, but this was focused on the `input` component. So it would be great if you create new issue which focuses on custom components which are required in a form (like `md-select`, `md-chips` - there you can use my second demo).

**Comment by DevVersion** [comment]
@bpaczkowski I've updated the codepen assets to use `1.0.6`.

The codepens show clearly how the select integrates the `required` attribute within the validation of a form.

- The `$valid` property will be always `false` until the user selects something in the `md-select`. (Works as expected)

- The `$valid` property will be `true` when the `md-select` doesn't have any value (Works as expected)

The form will be always submitted, regardless of the select, because you explicitly click on the submit button and the browser can't detect the `md-select` as an native form control. 

There is nothing we can do about native submit prevention.

**Comment by bpaczkowski** [comment]
Ahh, so that's the reason why it was still submitting. Thank you for clarifying this for me.

**Comment by caneraydinbey** [comment]
Doesnot work still. http://stackoverflow.com/questions/39018074/empty-value-of-select-box-doesnot-give-error-material-design

**Comment by adibas03** [comment]
Issue Fixed in version 1.13 https://github.com/angular/material/issues/8529

**Comment by AntonisFK** [comment]
Still having this issue on v1.1.4
